lib/plines/step.rb

```diff
@@ -135,13 +135,20 @@ def jobs_for(batch_data)
       end
     end
 
+    JobBatchCreationAborted = Class.new(StandardError)
+
     def perform(qless_job)
       batch = JobBatch.find(qless_job.client, pipeline,
                             qless_job.data.fetch("_job_batch_id"))
 
       if (retry_delay = batch.paused_retry_delay)
-        qless_job.move(qless_job.queue_name, delay: retry_delay)
-        return
+        if batch.creation_appears_to_be_stuck?
```

Comment:
So what if it isn't actually "stuck," but just taking longer than the timeout. Perhaps this job's perform method is called one second after the timeout, and two seconds after the timeout it finishes creating the job batch? Then wouldn't we observe a running, non-stuck job batch that can never complete because it has jobs that are in a failed state, having raised the `JobBatchCreationAborted` error?

I would hope that if we hit this timeout, there would be some handling to make sure that further creation of the job batch wouldn't happen so it can't finish creation.